tests/5.0/requires/test_requires_unified_shared_memory_heap_is_device_ptr.c

```diff
@@ -0,0 +1,71 @@
+//===---test_requires_unified_shared_memory.c -------------------------------===//
+//
+// OpenMP API Version 5.0 Nov 2018
+// 
+// This test Checks for unified shared memory of an array that is allocated on 
+// the heap and that is accessed from host and device with the same pointer
+//
+// To guarantee the use of the device pointer, we use the is_device_ptr clause
+//
+////===----------------------------------------------------------------------===//
+#include <omp.h>
+#include <stdio.h>
+#include <stdlib.h>
+#include "ompvv.h"
+
+#define N 1024
+
+#pragma omp requires unified_shared_memory
+
+int unified_shared_memory_heap() {
+  OMPVV_INFOMSG("Unified shared memory testing - Array on heap");
+  int errors = 0;
+  
+  int *anArray;
+  int anArrayCopy[N];
+
+  anArray = (int*)malloc(sizeof(int)*N);
+
+  for (int i = 0; i < N; i++) {
+    anArray[i] = i;
+    anArrayCopy[i] = 0;
+  }
+  // Modify in the device
+#pragma omp target is_device_ptr(anArray)
```

Comment:
Need to check if this is permitted by spec? Can a host-malloced pointer be used in is_device_ptr?

Reply:
If it's the host device, then sure.  The requirement is that it's a valid host-handle to data for the device in question, for the host that should generally be a regular host pointer (exceptions might be opencl or sycl backends with certain properties)